### PR TITLE
NavigationThreadingOptimizations STABLE 100% 

### PR DESCRIPTION
**Pull request by atuchin-m**
For https://github.com/brave/brave-browser/issues/23267

### CODE REVIEW

**Review by iefremov — APPROVED** [review]

**Comment by Ashtonian** [comment]
Please merge.

**Review by kjozwiak — APPROVED** [review]
Uplift into `production` approved after deliberating with @brave/uplift-approvers 👍

**Pull request merged**
